exec/java-exec/src/main/java/org/apache/drill/exec/record/metadata/AbstractColumnMetadata.java

```diff
@@ -87,7 +86,8 @@ public AbstractColumnMetadata(AbstractColumnMetadata from) {
     expectedElementCount = from.expectedElementCount;
   }
 
-  protected void bind(TupleSchema parentTuple) { }
+  @Override
+  public void bind(TupleMetadata parentTuple) { }
```

Comment:
Not sure if an issue here, but the "Schema"/"Metadata" split handles the fact that the column accessors use these interfaces, but are in the vector package. The "Metadata" interfaces thus are designed to be visible to vector. On the other hand, the implementation needed visibility to BatchSchema, which was defined in java-exec.

So, some careful fiddling was needed to ensure that the column accessors could work just with the interfaces, leaving the implementations to be defined in java-exec.

All this can certainly be fixed; just tried to minimize code impact during the long merge process.

Anyway, the change above may not be an issue, but you now have the context to understand if it is.